content/ADR-189-social-service-M2.md

```diff
@@ -0,0 +1,131 @@
+
+## Solution Space Exploration
+
+There are two main approaches:
+
+### Websocket / Long polling
+
+Websocket inherently comes with the benefit of establishing a bidirectional connection, enabling more features for the future (i.e. presence). Another big plus of using websocket is that the client can leverage protobuff and RPC for communication between the server and the client.
+On the other hand, Websocket + RPC implies that the Social server will need to be the first implementation of RPC-rust which is not yet a production tested product.
+
+### Server Sent Event
+
+This solution doesn't imply a constant connection between the server and the client and allows the existance of a notification system, it's cheap for the server and the client. The biggest problem with this solution is that there isn't an existent transport in unity for SSE, and in the future when presence is implemented there might need to be a big refactor and a replacement of the technology.
+
+## Specification
+
+Due to the benefits explained over the SSE solution, Websocket is the solution that will be further explored in this document.
+
+The new flow for a friendship will be the following:
+
+```mermaid
+sequenceDiagram
+    participant User 1
+    participant User 2
+    participant Social service
+    participant Matrix
+
+    User 1 ->> Social service: Add friend (new friend request)
+    Social service ->>  User 2: New request from user 1
+    User 2 ->> Social service: Accept friendship
+    Social service ->> Matrix: create room
+    Social service -->>+ User 1: New FriendshipId: roomId
+    Social service -->>+ User 2: New FriendshipId: roomId
+```
+
+This implies that rooms in Matrix will not be created until the friendship is established. This does not make the opposite true, if the friendship ends the room won't be destroyed (to prevent the history from being lost).
+
+Now, when a user logs in to decentraland, the friendship requests will also be obtained from the social service instead of the matrix sdk.
+
+First, the client will need to login via HTTP, then establish a connection with the server, once correctly established, the server will then send two messages one with the current friends and another with the pending friend requests.
+
+```mermaid
+sequenceDiagram
+    participant User 1
+    participant Social service
+    participant Matrix
+
+    User 1 ->>+ Social service: Login
+    Social service ->>+ Matrix: Login
+    Matrix -->>- Social service: bearer token
+    Social service -->>- User 1: bearer token
+    User 1 ->>+ Social service: Connect (token)
+    Social service -->> User 1: Friends
+    Social service -->> User 1: Friend Requests
+    loop  Connection messages examples
+    User 1 ->> Social service: Friend event (ACCEPT, DELETE, etc.)
+    User 1 ->> Social service: Update position
```

Comment:
Why do we need to add the update position here? This seems to still be processed in Matrix FTBB.